shared/test/test_deliverer.rb

```diff
@@ -0,0 +1,48 @@
+require_relative 'test_helper'
+require 'cdo/poste'
+
+class FakeSmtp
+  def send_message(message, from_address, *to_addresses)
+    @message = message
+    @from_address = from_address
+    @to_addresses = to_addresses
+  end
+end
+
+class DelivererTest < Minitest::Test
+  def setup
+    @fake_smtp = FakeSmtp.new
+    Deliverer.any_instance.stubs(:reset_connection).returns(@fake_smtp)
+    @deliverer = Deliverer.new({})
+  end
+
+  def test_deliverer_render
+    template = @deliverer.load_template("dashboard")
+    header, html, text = template.render
+    assert_equal "Code.org <noreply@code.org>", header["from"]
+    refute_nil html
+    assert_nil text
+  end
+
+  def test_deliverer_send
+    email = "test@deliverer.send"
+    contact = Poste2.create_recipient(email, {ip_address: '5.6.7.8.'})
+
+    # Sequel doesn't have a "find or create by", so we implement it manually
+    message = POSTE_DB[:poste_messages].where(name: "dashboard").first
+    message_id = message.nil? ? POSTE_DB[:poste_messages].insert({name: "dashboard"}) : message[:id]
+
+    delivery = {
+      id: 1,
+      contact_id: contact[:id],
+      message_id: message_id,
+      params: {}.to_json
+    }
+    @deliverer.send(delivery)
+    message = @fake_smtp.instance_variable_get(:@message)
```

Comment:
It might read better to add
```rb
attr_reader :message, :from_address, :to_address
```
to `FakeSmtp`.  My only concern would be whether you think the test becomes misleading as documentation of how you'd typically use this class.

Reply:
I'll add a comment explaining why I'm doing that